api/src/main/java/io/minio/NotificationListener.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package io.minio;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.minio.notification.NotificationInfo;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Scanner;
+
+/**
+ * NotificationListener is used for listening notification.
+ */
+public class NotificationListener  {
+
+  Scanner sc = null;
+  String notificationString = "";
+  NotificationInfo ni = null;
+  Reader inputReader;
+  boolean isClosed = false;
+  ObjectMapper mapper = new ObjectMapper();
```

Comment:
1. use `reader`, `scanner`, `notificationInfo` instead of `inputReader`, `sc`, `ni`
2. group relevant fields together `reader`, `scanner` and `mapper`, `notificationInfo`, then `isClosed`